src/opnsense/www/js/opnsense.js

```diff
@@ -120,32 +125,32 @@ function getFormData(parent) {
  */
 function setFormData(parent,data) {
     $( "#"+parent+"  input,#"+parent+" select,#"+parent+" span,#"+parent+" textarea" ).each(function( index ) {
-        if ($(this).prop('id') == undefined || $(this).prop('id') == "") {
+        if ($(this).prop('id') === undefined || $(this).prop('id') === "") {
             // we need an id.
             return;
         }
         var node = data;
         var targetNode = $(this); // document node to fetch data to
         var keyparts = $(this).prop('id').split('.');
-        $.each(keyparts,function(indx,keypart){
+        $.each(keyparts,function(index, keypart){
             if (keypart in node) {
-                if (indx < keyparts.length - 1) {
+                if (index < keyparts.length - 1) {
                     node = node[keypart];
                 } else {
                     // data node found, handle per type
                     if (targetNode.is("select")) {
                         // handle select boxes
                         targetNode.empty(); // flush
                         $.each(node[keypart],function(indxItem, keyItem){
-                            var opt = $("<option>").val(indxItem).text(keyItem["value"]);
-                            if (keyItem["selected"] != "0") {
+                            var opt = $("<option>").val(htmlDecode(indxItem)).text(keyItem["value"]);
```

Comment:
seems to be working fine, thanks!

Reply:
If you don't like the rest, just take that line - This is important not to break the frontend for API values containing HTML characters.